pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3488,6 +3488,167 @@ func TestReconcileWithWhenExpressionsWithTaskResults(t *testing.T) {
 	}
 }
 
+func TestReconcileWithWhenExpressionsScopedToTask(t *testing.T) {
```

Comment:
nit: adding a graph in the comment above would be helpful to understand

Reply:
added a graph, so hard to draw them though, hope it helps!